### PR TITLE
LSP server request message is misinterpreted as a response message (not properly fixed by 9.0.1922)

### PR DESCRIPTION
**Pull request by yegappan**

### CODE REVIEW

File: src/optionstr.c [hunk 1]
```diff
@@ -1858,7 +1858,7 @@ did_set_isopt(optset_T *args)
  * The 'jumpoptions' option is changed.
  */
     char *
-did_set_jumpoptions(optset_T *args)
+did_set_jumpoptions(optset_T *args UNUSED)
```
Comment (Shane-XB-Qian) [inline src/optionstr.c:1861]: what's this? related?

File: src/channel.c [hunk 2]
```diff
@@ -2466,6 +2466,12 @@ channel_get_json(
 		d = item->jq_value->vval.v_dict;
 		if (d == NULL)
 		    goto nextitem;
+		// When looking for a response message from the LSP server,
+		// ignore new LSP request and notification messages.  LSP
+		// request and notification messages have the "method" field in
+		// the header and the response messages do not have this field.
+		if (dict_has_key(d, "method"))
+		    goto nextitem;
```
Comment (Shane-XB-Qian) [inline src/channel.c:2474]: is that really ok? 
in that reported case, seems if ignore or not handle that workspace/configure msg from lsp server, that server looks would be hanging there.

File: src/optionstr.c (hunk 1 above)
Comment (yegappan) [reply src/optionstr.c:1861]: I don't want to open another PR just to fix this compiler warning.

File: src/channel.c (hunk 2 above)
Comment (yegappan) [reply src/channel.c:2474]: Yes.  The request messages are not dropped, they will be processed after processing the response message.

Comment (Shane-XB-Qian) [reply src/channel.c:2474]: i felt it's not same behavior or worked one for that reported case.

**Comment by Shane-XB-Qian** [comment]
v9.0.1922 was helped that reported case, however looks this not.
@lithammer maybe you can try again at your side too.

**Comment by lithammer** [comment]
Yeah I still have the issue after this patch (tested on 1927).

```
Error: request textDocument/inlayHint failed (no result):
{'method': 'workspace/configuration', 'jsonrpc': '2.0', 'id': 2, 'params': {'items': [{'section': 'gopls'}]}}
```

**Pull request closed without merging**